src/features/session_replay/aggregate/index.js

```diff
@@ -122,8 +134,10 @@ export class Aggregate extends AggregateBase {
           this.mode = MODE.FULL
           // if the error was noticed AFTER the recorder was already imported....
           if (recorder && this.initialized) {
-            this.stopRecording()
-            this.startRecording()
+            if (globalScope?.document.visibilityState === 'visible') {
```

Comment:
There are a couple other things going on in this handler that doesn't make sense when page hidden. Is this awaiting feedback from Steve?

Reply:
We are awaiting Steve on some things yes but lets talk about the other things that should be short circuited if not visible